src/input/actions.rs

```diff
@@ -1107,11 +1133,13 @@ fn to_previous_workspace(
     workspace_state: &mut WorkspaceUpdateGuard<'_, State>,
 ) -> Result<Option<Point<i32, Global>>, InvalidWorkspaceIndex> {
     let current_output = seat.active_output();
-    let workspace = shell
-        .workspaces
-        .active_num(&current_output)
-        .1
-        .saturating_sub(1);
+    let workspace = shell.workspaces.active_num(&current_output).1;
+
+    if workspace == 0 {
```

Comment:
This is here as workspace is of type `usize` where `0usize.saturating_sub(1) results in 0` which isn't an invalid index. This is here to allow PreviousWorkspace action to fallback to SwitchOutput on error.